compiler/ballerina-lang/src/main/resources/grammar/BallerinaLexer.g4

```diff
@@ -963,36 +975,32 @@ DeprecatedValidCharSequence
 mode STRING_TEMPLATE;
 
 StringTemplateLiteralEnd
-    :   '`' { inTemplate = false; }          -> popMode
+    :   '`' { inStringTemplate = false; }          -> popMode
     ;
 
 StringTemplateExpressionStart
-    :   StringTemplateText? ExpressionStart            -> pushMode(DEFAULT_MODE)
+    :   StringTemplateText? InterpolationExpressionStart            -> pushMode(DEFAULT_MODE)
     ;
 
-// We cannot use "StringTemplateBracesSequence? (StringTemplateStringChar StringTemplateBracesSequence?)*" because it
-// can match an empty string.
 StringTemplateText
-    :   StringTemplateValidCharSequence? (StringTemplateStringChar StringTemplateValidCharSequence?)+
-    |   StringTemplateValidCharSequence (StringTemplateStringChar StringTemplateValidCharSequence?)*
+    :   StringTemplateValidCharSequence+ Dollar*
+    |   Dollar+
     ;
 
 fragment
-StringTemplateStringChar
-    :   ~[`{\\]
-    |   '\\' [`{]
-    |   WS
-    |   StringLiteralEscapedSequence
+Dollar
```

Comment:
Dollar -> DOLLAR
also should this be moved to top?

Reply:
Thought of it. However, `$` is expected to match only within the string template literal and we don't have any matching patterns within the `default mode`. Hence, I created it as a fragment within the `STRING_TEMPLATE` mode.